services/core/src/utils/CheckedContract.ts

```diff
@@ -253,10 +264,10 @@ export class CheckedContract {
  * @param log whether or not to log
  * @returns the fetched file if found; null otherwise
  */
-async function performFetch(
+export async function performFetch(
```

Comment:
I see the last change resolves my prev. comment of the case `hash === null`. But isn't the whole purpose of the function to do the fetch and compare with the hash? Without the hash it is no different than a standard `fetch()` it seems. See the other comment in `VerificationController`. 

Also currently the function comments do not reflect the functionality.

Is the only reason you added `hash` as optional is to use it in the case at `VerificationController:876`? I feel we should've kept this function as is and done a simple fetch at `VerificationController:876`. Maybe a more descriptive name the a simple `performFetch` would be better, something like `fetchAndCompareHash`.

Reply:
I think it's a way to always log when fetching something, use the timeout and log errors

Reply:
Ok, then just updating the comment accordingly should suffice